### PR TITLE
Fixed growth of final deduplication partition size due to locking all partitions

### PR DESCRIPTION
**Pull request by penguinlav**
### Describe Your Changes

Situation. VM cluster with 6 months of metrics and minScrapeInterval set to 1 sec. Disk space is getting closer and closer to 70%. It was decided to increase minScrapeInterval to 30 sec.

After setting these settings and rebooting vmstorage, disk space suddenly started to run out. From the logs everything looks fine:
```
info        VictoriaMetrics/lib/storage/partition.go:1335        start removing duplicate samples from partition (/var/lib/victoriametrics/data/big/2024_03, /var/lib/victoriametrics/data/small/2024_03)
info        VictoriaMetrics/lib/storage/partition.go:1479        merged (33 parts, 139203693984 rows, 76745971 blocks, 82319701917 bytes) into (1 part, 14123920969 rows, 44441559 blocks, 16361130334 bytes) in 2202.133 seconds at 63213123 rows/sec to "/var/lib/victoriametrics/data/big/2024_03/17F2AF621995BDF4"
info        VictoriaMetrics/lib/storage/partition.go:1339        duplicate samples have been removed from partition (/var/lib/victoriametrics/data/big/2024_03, /var/lib/victoriametrics/data/small/2024_03) in 2202.133 seconds
```
partitions are merging, `duplicate samples have been removed from partition`. But free space is monotonously running out and becoming closer to RO mode:
image

At the same time, the VM is freeing up memory, probably because it stops using the partitions that were merged.
image

Long story short it seems that during the final deduplication, all partitions are locked and cannot be deleted until the entire final deduplication process of all partitions is completed.

### Checklist

The following checks are **mandatory**:

- [x] My change adheres [VictoriaMetrics contributing guidelines](https://docs.victoriametrics.com/contributing/).

### CODE REVIEW

**Comment by penguinlav** [comment]
My fault, everything is fine there. I read the graphs incorrectly.
